pxr/usd/usd/testenv/testUsdPrims.py

```diff
@@ -614,13 +614,19 @@ def test_PropertyOrder(self):
         assert po
         attrs = po.GetAttributes()
         # expected order:
-        expected = ['A0', 'a1', 'a2', 'A3', 'a4', 'a5', 'a10', 'A20']
+        if Tf.GetEnvSetting('TF_UTF8_IDENTIFIERS'):
+            expected = ['A0', 'a1', 'a10', 'a2', 'A20', 'A3', 'a4', 'a5']
```

Comment:
This will need to be addressed -- it's important to us that sequences of numerals sort as their numeric value.

Reply:
See comment above. I think it would be sufficient to only work with ASCII digits this way and it could be merged with necessary changes so that ASCII punctuation is considered more significant than later letters. This is done by extracting the ASCII non-alphabetic sequences and comparing them with our own code.

Reply:
Implementation now uses current sorting algorithm with the addition of UTF-8 byte sorting when there is a mismatch.